lms/views/api/canvas/authorize.py

```diff
@@ -0,0 +1,48 @@
+import secrets
+from urllib.parse import urlencode, urlparse, urlunparse
+
+from pyramid.httpexceptions import HTTPFound
+from pyramid.view import view_config
+
+from lms.validation import CANVAS_OAUTH_CALLBACK_SCHEMA
+
+
+@view_config(route_name="canvas_api_authorize", request_method="GET")
+def authorize(request):
+    ai_getter = request.find_service(name="ai_getter")
```

Comment:
I had to look up what `ai_getter` was while reading this. In retrospect it would have been better to use a more descriptive name when registering the service.